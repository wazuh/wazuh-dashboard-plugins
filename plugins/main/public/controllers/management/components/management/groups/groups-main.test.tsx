/*
 * Wazuh app - React test for Group-Main component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import WzGroups from './groups-main';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.mock('../../../../../kibana-services', () => ({
  getAngularModule: jest.fn(),
  getHttp: () => ({
    basePath: {
      prepend: str => str,
    },
  }),
}));

const mockProps = {
  section: 'groups',
  configurationProps: {
    agent: {
      id: '000',
    },
  },
  logtestProps: {
    showClose: true,
    onFlyout: true,
  },
  state: {
    section: '',
  },
};
const mockStore = configureMockStore();
const store = mockStore({});

describe('Group main component', () => {
  it('renders correctly to match the snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <WzGroups {...mockProps} />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
