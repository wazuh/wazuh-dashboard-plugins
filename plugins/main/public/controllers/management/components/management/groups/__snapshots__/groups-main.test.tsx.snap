// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group main component renders correctly to match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(Component)
    configurationProps={
      Object {
        "agent": Object {
          "id": "000",
        },
      }
    }
    logtestProps={
      Object {
        "onFlyout": true,
        "showClose": true,
      }
    }
    section="groups"
    state={
      Object {
        "section": "",
      }
    }
  />
</ContextProvider>
`;
